llvm/lib/Linker/IRMover.cpp

```diff
@@ -297,7 +297,7 @@ Type *TypeMapTy::get(Type *Ty, SmallPtrSet<StructType *, 8> &Visited) {
     return *Entry = VectorType::get(ElementTypes[0],
                                     cast<VectorType>(Ty)->getElementCount());
   case Type::PointerTyID:
-    return *Entry = PointerType::get(ElementTypes[0],
+    return *Entry = PointerType::get(Ty->getContext(),
                                      cast<PointerType>(Ty)->getAddressSpace());
```

Comment:
I think this case can be dropped entirely, it should now be handled by the generic code.

Reply:
Yeah seems reasonable